src/android/InAppBrowser.java

```diff
@@ -1505,16 +1550,11 @@ public void onReceivedSslError(WebView view, SslErrorHandler handler, SslError e
         @Override
         public void onReceivedHttpAuthRequest(WebView view, HttpAuthHandler handler, String host, String realm) {
             // Check for basicAuthSettings that match the host
-            if (basicAuthLogins != null) {
-                for (HashMap.Entry<String, BasicAuthLogin> entry : basicAuthLogins.entrySet()) {
-                    String loginhost = entry.getKey();
-                    BasicAuthLogin login = entry.getValue();
-                    if (loginhost.equals(host)) {
-                        LOG.i(LOG_TAG, "onReceivedHttpAuthRequest - found user/pass for matching host:" + host);
-                        handler.proceed(login.user, login.pass);
-                        return;
-                    }
-                }
+            if (basicAuthLogins != null && basicAuthLogins.get(host) != null) {
```

Comment:
Fixed this code for basicAuth that was never converted to use the Map lookup but was still looping through keys.